src/nvim/eval.c

```diff
@@ -4239,11 +4239,17 @@ static int eval7(
         // use its contents.
         s = deref_func_name((const char *)s, &len, &partial, !evaluate);
 
+        // Need to make a copy, in case evaluating the arguments makes
+        // the name invalid.
+        s = vim_strsave(s);
```

Comment:
We probably only do need `len` of `s`. It seems that the only place where entire `s` is used is when reporting an error in `get_func_tv`. Copying only `len` of `s` might even make the error text more useful.